plugins/src/signaller/imp.rs

```diff
@@ -28,17 +28,51 @@ struct State {
     receive_task_handle: Option<task::JoinHandle<()>>,
 }
 
+#[derive(Serialize, Deserialize, Debug, PartialEq)]
```

Comment:
Might want to move the message structs into a separate module

Reply:
done